### PR TITLE
MGMT-8498: fixing issue with not mounting pki due to old next_step_runner code.

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-8498](https://issues.redhat.com/browse/MGMT-8498): fixing issue with not mounting pki due to old next_step_runner code.
Removig v2_next_step_runner_cmd.go as it is duplicate of
next_step_runner_cmd.go and changing ext_step_runner_cmd.go to use infra
env. This will allow us to have one file with better name and will allow
not to make mistakes cause of code duplication

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by filanov** [comment]
@tsorya  please make sure to fill in the PR template.

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -3715,15 +3715,15 @@ func (b *bareMetalInventory) generateV2NextStepRunnerCommand(ctx context.Context
 			params.NewHostParams.HostID.String(), params.InfraEnvID.String(), params.NewHostParams.DiscoveryAgentVersion, b.AgentDockerImg)
 	}
 
-	config := hostcommands.V2NextStepRunnerConfig{
+	config := hostcommands.NextStepRunnerConfig{
```
Comment (filanov) [inline internal/bminventory/inventory.go:3718]: Why did you remove the v2? v1 will be deprecated and removed

**Comment by filanov** [comment]
cc: @yevgeny-shnaidman

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (tsorya) [reply internal/bminventory/inventory.go:3718]: This is v2 , i just changed the name to the normal one. There is only one file now so no need to call it v2 or any other version

Comment (filanov) [reply internal/bminventory/inventory.go:3718]: I'm not sure if the agent will know how to parser it. isn't it a different struct?

Comment (tsorya) [reply internal/bminventory/inventory.go:3718]: agent will not get this struct as it is internal one, agent will get next_step struct

Comment (tsorya) [reply internal/bminventory/inventory.go:3718]: this is internal struct we can call it what we want, it is not relevant to api itself, all the change that was done for v2 was adding infraenv-id and not cluster.

Comment (filanov) [reply internal/bminventory/inventory.go:3718]: OK, thanks for the explanation
